admin/src/views/Analytics.vue

```diff
@@ -0,0 +1,293 @@
+      const index = legendItem.datasetIndex;
+      const ci = this.chart;
+      const size = ci.data.datasets.length;
+      for (let i = 0; i < size; i += 1) {
+        if (i !== index) {
+          const meta = ci.getDatasetMeta(i);
+          meta.hidden = meta.hidden === null ? !ci.data.datasets[i].hidden : null;
+        }
+      }
+      // rerender the chart
+      ci.update();
+    };
+    return {
+      agentName: this.$route.params.agentName,
+      textMsg: '',
+      classMsg: '',
+      analyticsData: '',
+      period: 'Last 30 days',
+      doghnutPeriod: 'Last day',
+      chartOptions: {
+        responsive: true,
+        maintainAspectRatio: false,
+        scales: {
+          yAxes: [{
+            ticks: {
+              beginAtZero: true,
+            },
+          }],
+        },
+      },
+      chartDataLineIntentsOptions: {
+        responsive: true,
+        maintainAspectRatio: false,
+        scales: {
+          yAxes: [{
+            display: false,
+            ticks: {
+              stepSize: 1,
+              beginAtZero: true,
+            },
+          }],
+        },
+        legend: {
+          onClick: handleIntentLegendClick,
+        },
+        title: {
+          display: true,
+          text: 'Intents',
+          fontSize: 15,
+          fontStyle: 'normal',
+        },
+      },
+      doghnutOptions: {
+        responsive: true,
+        maintainAspectRatio: false,
+        legend: {
+          position: 'right',
+        },
+        title: {
+          display: true,
+          text: 'Intents Doghnut',
+          fontSize: 15,
+          fontStyle: 'normal',
+        },
+      },
+      dateFields: '',
+      trafficFields: '',
+      uniqueUsersFields: '',
+      fallbackFields: '',
+      intentFields: [],
+      countFields: Array<any>(),
+      allDateFields: '',
+      allTrafficFields: '',
+      allUniqueUsersFields: '',
+      allFallbackFields: '',
+    };
+  },
+  computed: {
+    chartDataLineTraffic(): any {
+      return {
+        labels: this.dateFields,
+        datasets: [
+          {
+            label: 'traffic',
+            data: this.trafficFields,
+            fill: false,
+            borderColor: 'rgb(75, 192, 192)',
+            tension: 0.1,
+          },
+        ],
+      };
+    },
+    chartDataLineUniqueUsers(): any {
+      return {
+        labels: this.dateFields,
+        datasets: [
+          {
+            label: 'unique_users',
+            data: this.uniqueUsersFields,
+            fill: false,
+            borderColor: 'rgb(75, 192, 192)',
+            tension: 0.1,
+          },
+        ],
+      };
+    },
+    chartDataLineIntents(): any {
+      const limit = this.period === 'Last 7 days' ? 7 : 30;
+      const limitedAnalyticsData = _.takeRight((this.analyticsData as any).analytics, limit);
+      const flatAnalyticsData = _.flatMap(limitedAnalyticsData, 'intents_count');
```

Comment:
lot of constants ! you use chaining
you can do that:

const intentsDatasets = _(limitedAnalyticsData)
        .flatMap('intents_count')
        .uniqBy('intent')
        .map((item: any) => ({
          label: item.intent,
          data: this.getCountsOfIntent(item.intent, limitedAnalyticsData),
          fill: false,
          borderColor: this.getRandomColor(),
          tension: 0.1,
        }))
        .value();